asllib/Interpreter.ml

```diff
@@ -65,35 +67,113 @@ module Make (B : Backend.S) (C : Config) = struct
 
   type env = IEnv.env
 
+  type value_read_from = B.value * identifier * scope
+
+  type 'a maybe_exception =
+    | Normal of 'a
+    | Throwing of (value_read_from * ty) option * env
+
+  (** An intermediate result of a statement. *)
+  type control_flow_state =
+    | Returning of B.value list * IEnv.global
+        (** Control flow interruption: skip to the end of the function. *)
+    | Continuing of env  (** Normal behaviour: pass to next statement. *)
+
+  type expr_eval_type = (B.value * env) maybe_exception m
+  type stmt_eval_type = control_flow_state maybe_exception m
+  type func_eval_type = (value_read_from list * IEnv.global) maybe_exception m
+
+  (*****************************************************************************)
+  (*                                                                           *)
+  (*                           Monadic operators                               *)
+  (*                                                                           *)
+  (*****************************************************************************)
+
   let one = B.v_of_int 1
+
+  (* Return *)
+  (* ------ *)
   let return = B.return
 
-  (* Binding operators, first with data *)
+  let return_normal v = Normal v |> return
+
+  let return_continue env : stmt_eval_type = Continuing env |> return_normal
+
+  let return_return env vs : stmt_eval_type =
+    Returning (vs, env.IEnv.global) |> return_normal
+
+  (* Bind *)
+  (* ---- *)
+
+  (* Sequential bind *)
+  let ( let*| ) = B.bind_seq
+
+  (* Data bind *)
   let ( let* ) = B.bind_data
   let ( >>= ) = B.bind_data
 
-  (* Then with po *)
-  let ( let*| ) = B.bind_seq
+  (* Parallel bind *)
+  let ( and* ) = B.bind_par
+  (* could and* be called par? *)
```

Comment:
We can rename `bind_par` to be called `par`, but to use the syntactic feature of the `and*`, we can only name it something that is formed by "and" followed by some operators (see [the manual](https://v2.ocaml.org/manual/bindingops.html#and-operator))
If we dropped the syntactic sugar of the `and*`, then why not name it `bind_par` (or its name in `Backend`) not to confuse people?